pkg/apis/options/login_url_parameters.go

```diff
@@ -0,0 +1,59 @@
+package options
+
+import (
+	"regexp"
+
+	"github.com/oauth2-proxy/oauth2-proxy/v7/providers"
```

Comment:
This is likely to cause a dependency loop in the future. This package must not import anything else from within the project

Reply:
The options package already imports the providers package in a couple of places, `legacy_options.go` and `providers.go` just import it for a couple of constants but `options.go` is a harder dependency (one of the "internal values set after validation" is a `providers.Provider`).

Reply:
This came up yesterday while reviewing another PR, I'm working on something to resolve this issue which I hope to post within the next day or 2